worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapPartitionDataWriter.java

```diff
@@ -63,19 +65,25 @@ public MapPartitionDataWriter(
     super(storageManager, workerSource, conf, deviceMonitor, writerContext, false);
 
     Preconditions.checkState(diskFileInfo != null);
-    if (!diskFileInfo.isHdfs()) {
+    if (!diskFileInfo.isDFS()) {
       indexChannel = FileChannelUtils.createWritableFileChannel(diskFileInfo.getIndexPath());
     } else {
+      FileSystem hadoopFs = null;
```

Comment:
A map partition data writer is connected to one partition location which means that its location is fixed. So cache the fs and avoid the unnecessary condition checks.